### ISSUE TITLE
Network namespaces?

### ISSUE THREAD

**Issue by vaenkyrae** [evilsocket/opensnitch#378]
Hello, I run some applications in a seperate namespace using

$ sudo ip netns exec [namespace] sudo -u [user] [command]

These connections are not visible in opensnitch (or other network monitoring tools unless ran with the above launcher). Could it be possible for opensnitch to intercept them?

**Comment by gustavo-iniguez-goya** [comment]
Hi @vaenkyrae ,

Yes, at the moment we don't intercept those connections. We only intercept connections in the OUTPUT chain, and those connections travel through the FORWARD chain.

In order to intercept them we need an additional rule (even using ebpf):
`sudo iptables -t mangle -I FORWARD -j NFQUEUE --queue-num 0 --queue-bypass`

And you will start seeing some logs:
```
[2021-03-24 14:05:30]  DBG  new connection tcp => 46344:192.168.2.2 -> 8.8.8.8:23 uid: 4294967295
[2021-03-24 14:05:30]  DBG  netlink socket error: Warning, no message nor error from netlink - 46344:192.168.2.2 -> 8.8.8.8:23
[2021-03-24 14:05:30]  DBG  Searching for tcp6 netstat entry instead of tcp
[2021-03-24 14:05:30]  DBG  <== no inodes found, applying default action.
```

The inode of the connection is not found in the kernel, so probably we'll need to get it querying for the namespaces or something like that. I'll investigate it.

**Comment by gustavo-iniguez-goya** [comment]
~we'll need to get it querying for the namespaces or something like that. I'll investigate it.~
`socket_diag shows information only about sockets from a namespace where
a diag socket lives.`
https://git.ustc.edu.cn/chendotjs/linux/-/blob/v4.15/net/unix/diag.c

**Comment by gustavo-iniguez-goya** [comment]
this is also fixed with latest changes.

**Comment by gustavo-iniguez-goya** [comment]
hey @vaenkyrae , there's a new release with eBPF support that should fix this case, could you install it and follow these instructions? https://github.com/evilsocket/opensnitch/issues/385#issuecomment-844371925

**Comment by gustavo-iniguez-goya** [comment]
this issue should be fixed with  v1.4.0, eBPF as process monitor method and the firewall rule mentioned in the above comment.

**Comment by NDagestad** [comment]
Hi, given that this has been fixed by switching to eBPF, the is the recommendation of the wiki to disable unprivileged network namespaces still valid?
